dandischema/digests/dandietag.py

```diff
@@ -0,0 +1,202 @@
+# Derived from <https://github.com/girder/django-s3-file-field/blob/master/
+# s3_file_field/_multipart.py>, copyright Kitware, Inc. <kitware@kitware.com>
+# under the Apache 2.0 license
+
+from dataclasses import dataclass
+from hashlib import md5
+import math
+import os
+from typing import Iterator, List, NamedTuple, Optional, Union
+
+
+def mb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 20
+
+
+def gb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 30
+
+
+def tb(bytes_size: int) -> int:
+    return bytes_size * 2 ** 40
+
+
+class Part(NamedTuple):
+    number: int
+    offset: int
+    size: int
+
+
+@dataclass
+class PartGenerator:
+    part_qty: int
+    initial_part_size: int
+    final_part_size: int
+
+    # S3 multipart limits: https://docs.aws.amazon.com/AmazonS3/latest/dev/qfacts.html
+    # 10k is the maximum number of allowed parts allowed by S3
+    MAX_PARTS = 10_000
+    # 5MB is the minimum part size allowed by S3
+    MIN_PART_SIZE = mb(5)
+    # 5GB is the maximum part size allowed by S3
+    MAX_PART_SIZE = gb(5)
```

Comment:
this is the one piece of info that gives me pause in terms of what amazon is accepting, we should check on our large files what the size of each part is.  because 5 GiB > 5 GB, so if 5GB is the limit then i would assume 5 GiB would not be accepted.

Reply:
would indeed be nice to clear it up, but I would keep it *as is* for the purpose of this PR to provide as smooth (no side-effects) transition from dandi-cli to dandischema

Reply:
i'll file an issue. this can be merged without these changes.